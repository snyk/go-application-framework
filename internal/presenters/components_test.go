package presenters

import (
	"fmt"
	"testing"

	"github.com/charmbracelet/lipgloss"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/muesli/termenv"
	"github.com/snyk/error-catalog-golang-public/snyk"
	"github.com/stretchr/testify/assert"
)

func Test_RenderError(t *testing.T) {
	for _, severity := range []string{"warn", "error", "fatal"} {
		t.Run(
			fmt.Sprintf("colors for severity %s", severity), func(t *testing.T) {
				err := snyk.NewTooManyRequestsError("")
				err.Level = severity
				lipgloss.SetColorProfile(termenv.TrueColor)
				output := RenderError(err)
				snaps.MatchSnapshot(t, output)

				lipgloss.SetColorProfile(termenv.TrueColor)
				lipgloss.SetHasDarkBackground(true)
				outputDark := RenderError(err)
				snaps.MatchSnapshot(t, outputDark)
			})
	}

	t.Run("without links", func(t *testing.T) {
		lipgloss.SetColorProfile(termenv.TrueColor)
		lipgloss.SetHasDarkBackground(false)
		output := RenderError(snyk.NewBadRequestError("A short error description"))

		assert.NotContains(t, output, "Help:")
		snaps.MatchSnapshot(t, output)
	})

	t.Run("with links", func(t *testing.T) {
		lipgloss.SetColorProfile(termenv.TrueColor)
		lipgloss.SetHasDarkBackground(false)
		output := RenderError(snyk.NewServerError("An error"))

		assert.Contains(t, output, "Help:")
		snaps.MatchSnapshot(t, output)
	})
}
