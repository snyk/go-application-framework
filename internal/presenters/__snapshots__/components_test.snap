
[Test_RenderError/colors_for_severity_error - 1]

[41m [0m[97;41mERROR[0m[41m [0m  [1mService temporarily throttled (SNYK-0001)[0m
HTTP:    429 
---

[Test_RenderError/colors_for_severity_error - 2]

[41m [0m[97;41mERROR[0m[41m [0m  [1mService temporarily throttled (SNYK-0001)[0m
HTTP:    429 
---

[Test_RenderError/colors_for_severity_fatal - 1]

[41m [0m[97;41mFATAL[0m[41m [0m  [1mService temporarily throttled (SNYK-0001)[0m
HTTP:    429 
---

[Test_RenderError/colors_for_severity_fatal - 2]

[41m [0m[97;41mFATAL[0m[41m [0m  [1mService temporarily throttled (SNYK-0001)[0m
HTTP:    429 
---

[Test_RenderError/colors_for_severity_warn - 1]

[43m [0m[30;43mWARN[0m[43m [0m   [1mService temporarily throttled (SNYK-0001)[0m
HTTP:    429 
---

[Test_RenderError/colors_for_severity_warn - 2]

[43m [0m[30;43mWARN[0m[43m [0m   [1mService temporarily throttled (SNYK-0001)[0m
HTTP:    429 
---

[Test_RenderError/with_links - 1]

[41m [0m[97;41mERROR[0m[41m [0m  [1mRequest not fulfilled due to server error (SNYK-9999)[0m
Info:    An error                                                                       
HTTP:    500 
Help:    https://status.snyk.io/                                                                                                       
         https://docs.snyk.io/getting-started/supported-languages-frameworks-and-feature-availability-overview#code-analysis-snyk-code 
---

[Test_RenderError/without_links - 1]

[41m [0m[97;41mERROR[0m[41m [0m  [1mClient request cannot be processed (SNYK-0003)[0m
Info:    A short error description                                                      
HTTP:    400 
---
